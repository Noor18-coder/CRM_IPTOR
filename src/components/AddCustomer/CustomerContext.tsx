import * as React from 'react';

export interface CustomerContextInterface {
  containerType: string;
  containerData?: any;
}

export const Context = React.createContext<CustomerContextInterface | null>(null);

export const CustomerContextProvider = Context.Provider;

export const CustomerContextConsumer = Context.Consumer;
