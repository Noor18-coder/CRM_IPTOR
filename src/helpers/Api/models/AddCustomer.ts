import { ErrorMessage } from './Approvals';

export interface AddBusinessPartnerDefaultParams {
  area?: string;
  country?: string;
  language?: string;
  primaryCurrency?: string;
  name?: string;
  addressLine1?: string;
  type?: number;
  phone?: string;
  EMAIL?: string;
  active?: boolean;
  businessPartner?: string;
}

export interface AddBusinessPartnerResponse {
  data: {
    businessPartner: string;
  };
  error?: string;
  messages?: ErrorMessage[];
}
