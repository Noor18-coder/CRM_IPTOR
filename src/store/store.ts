import { applyMiddleware, combineReducers, createStore, Store, Reducer } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storageSession from 'redux-persist/lib/storage/session'
import { composeWithDevTools } from 'redux-devtools-extension';

import { authReducer } from './Auth/Reducers';
import opportunityReducer from './Opportunity/Reducers';
import { OpportunityState } from './Opportunity/Types';
import businessPartnerReducer from './Customer/Reducer';
import { BusinessPartnerState } from './Customer/Types';
import { AuthState, AuthActions } from './Auth/Types';
import usersReducer from './Users/Reducers';
import { SaveUserAction, UsersData } from './Users/Types';
import { AppLoadingActions , InitialConfigState } from './InitialConfiguration/Types';
import configReducer from './InitialConfiguration/Reducers';
import {   AddOpportunityState } from './AddOpportunity/Types';
import addOpportunityReducer from './AddOpportunity/Reducers';

const persistConfig = {
  key: 'root',
  storage: storageSession,
  whitelist: ['auth','enviornmentConfigs', 'addOpportunity'] // which reducer want to store
};

export type AppActions = AuthActions;

export interface AppState {
  readonly auth : AuthState,
  readonly users: UsersData,
  readonly opportunities: OpportunityState,
  readonly businesspartners: BusinessPartnerState,
  readonly enviornmentConfigs:InitialConfigState,
  readonly addOpportunity: AddOpportunityState
}
const appReducer = combineReducers({
  auth: authReducer,
  users: usersReducer,
  opportunities: opportunityReducer,
  businesspartners: businessPartnerReducer,
  enviornmentConfigs: configReducer,
  addOpportunity: addOpportunityReducer
});

const rootReducer = (state: any, action: AppActions) => {
  if (action.type === 'LOGOUT_SUCCESS') {
    state = undefined
  }

  return appReducer(state , action)
}

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, composeWithDevTools(applyMiddleware(thunk)));
const persistor = persistStore(store);

export {
  store, persistor
};