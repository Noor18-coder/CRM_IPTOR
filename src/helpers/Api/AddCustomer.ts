import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { AddBusinessPartnerDefaultParams, AddBusinessPartnerResponse, SaveUserDefinedFieldParam } from './models';
import { ApiRequest } from './ApiRequest';

export default class AddCustomerApi {
  /** API Method */
  private static apiMethod = 'crmBusinessPartner.add';

  private static updateApiMethod = 'crmBusinessPartner.update';

  private static saveAttributeValueMethod = 'mopAttribute.add';

  private static saveAttributeValueMethodFile = 'SRONAM';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: crmBusinessPartner.add
   * */
  static async add(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.apiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async update(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.updateApiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async addAttributes(businessPartnerId: string, attributeType: string, attributeValue: string | number): Promise<AddBusinessPartnerResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: businessPartnerId,
      attributeValue,
    };
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }
}
