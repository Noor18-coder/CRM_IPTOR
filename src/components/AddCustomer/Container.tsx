import React from 'react';
import { useMediaQuery } from 'react-responsive';
import { useSelector } from "react-redux";

import Drawer from '@material-ui/core/Drawer';
import AddCustomer from './AddCustomer';
import EditCustomer from './EditCustomer';
import Loader from '../Shared/Loader/Loader';
import { AppState } from "../../store/store";
import { CustomerContextProvider } from './CustomerContext';

export interface containerProps {
    containerType: string;
    containerData?: any;
    groupType?: string;
    contactId?: string;
}

const Container: React.FC<containerProps> = ({ containerType, containerData, groupType, contactId }) => {

  const state:AppState = useSelector((state: AppState) => state);

  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 })
  
  return (
      <React.Fragment>
          <CustomerContextProvider value={containerData}>
              {state.addBusinessPartner.loader ? <Loader /> : null}
              {isMobile || isTablet ? (state.addBusinessPartner.addBusinessPartnerWindowActive ?
                  containerType === 'add' ? <AddCustomer /> :
                  containerType === 'edit' && < EditCustomer groupType={groupType} contactId={contactId} /> :
                  null) :
                  <Drawer anchor={'right'} open={state.addBusinessPartner.addBusinessPartnerWindowActive}>
                      {containerType === 'add' && <AddCustomer />}
                      {containerType === 'edit' && <EditCustomer groupType={groupType} contactId={contactId} />}
                  </Drawer>}
          </CustomerContextProvider>
    </React.Fragment>
  );
}

export default Container;