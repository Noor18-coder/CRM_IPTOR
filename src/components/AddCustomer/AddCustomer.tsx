import React from 'react';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router';

import i18n from '../../i18n';
import { AppState } from '../../store/store';
import ImageConfig from '../../config/ImageConfig';
import { emailPattern, numberPattern } from '../../helpers/utilities/lib';

import { AddBusinessPartnerDefaultParams, UserDefinedFieldReduxParams, CountryInfo, AreaInfo } from '../../helpers/Api/models';
import { setBusinessPartnerWindowActive, setBusinessPartnerLoader, resetBusinessPartnerData } from '../../store/AddCustomer/Actions';
import AddCustomerApi from '../../helpers/Api/AddCustomer';

const AddCustomer: React.FC = () => {
  const state: AppState = useSelector((addCustomerState: AppState) => addCustomerState);
  const dispatch: Dispatch<any> = useDispatch();
  const [customerFields, setCustomerFields] = React.useState<AddBusinessPartnerDefaultParams>({});
  const [attributes, setAttributes] = React.useState<UserDefinedFieldReduxParams[]>([]);
  const [isSubmit, setIsSubmit] = React.useState<boolean>(false);
  const [emailErr, setEmailErr] = React.useState<boolean>(false);
  const [phoneErr, setPhoneErr] = React.useState<boolean>(false);
  const history = useHistory();

  const onInputValueChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
    const { value, id } = e.currentTarget;
    const inputElement = document.getElementById(id) as HTMLInputElement;
    if (value.length && id === 'EMAIL' && !value.match(emailPattern)) {
      inputElement.style.border = '1px solid #ED2024';
      setIsSubmit(false);
      setEmailErr(true);
    } else if (value.length && id === 'phone' && !value.match(numberPattern)) {
      inputElement.style.border = '1px solid #ED2024';
      setIsSubmit(false);
      setPhoneErr(true);
    } else {
      inputElement.style.border = '1px solid #DAE2E7';
      setEmailErr(false);
      setPhoneErr(false);
    }
    setCustomerFields({
      ...customerFields,
      [e.currentTarget.id]: e.currentTarget.value,
    });
    if (e.currentTarget.id === 'EMAIL') {
      const newArray = [...attributes];
      newArray[0].attributeValue = e.currentTarget.value;
      setAttributes(newArray);
    }
  };

  const validateField = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
    const element = document.getElementById(e.currentTarget.id) as HTMLInputElement;
    if (e.currentTarget.value === '' || e.currentTarget.value.includes('Select')) {
      element.style.border = '1px solid #ED2024';
      setIsSubmit(false);
    } else {
      element.style.border = '1px solid #DAE2E7';
    }
  };

  const createCustomer = async () => {
    dispatch(setBusinessPartnerLoader(true));
    customerFields.primaryCurrency = state.enviornmentConfigs.defaultOpprtunityInfo.currencyLDA;
    customerFields.language = state.enviornmentConfigs.defaultOpprtunityInfo.language;
    customerFields.type = 1;
    if (!!state.auth.user.role && state.auth.user.role === 'Admin') customerFields.active = true;
    else customerFields.active = false;

    const data = await AddCustomerApi.add(customerFields);
    const customerId = data.data.businessPartner;

    Promise.all(
      attributes.map((obj: UserDefinedFieldReduxParams) => {
        return AddCustomerApi.addAttributes(customerId, obj.attributeType, obj.attributeValue);
      })
    ).then((details) => {
      return details;
    });

    dispatch(setBusinessPartnerWindowActive(false));
    dispatch(setBusinessPartnerLoader(false));
    dispatch(resetBusinessPartnerData());
    document.body.classList.remove('body-scroll-hidden');

    if (customerId) {
      history.push({ pathname: '/cust-details', state: { custId: customerId } });
    }
  };

  const onSubmit = () => {
    createCustomer();
  };

  const closeAction = () => {
    dispatch(resetBusinessPartnerData());
    dispatch(setBusinessPartnerWindowActive(false));
    dispatch(setBusinessPartnerLoader(false));
    document.body.classList.remove('body-scroll-hidden');
  };

  React.useEffect(() => {
    const emailTypeAttribute: UserDefinedFieldReduxParams = { attributeType: 'EMAIL', attributeValue: '' };
    setAttributes([emailTypeAttribute]);
  }, []);

  React.useEffect(() => {
    if (
      customerFields?.name &&
      customerFields?.addressLine1 &&
      customerFields?.country &&
      customerFields?.EMAIL &&
      customerFields?.area &&
      customerFields?.phone &&
      !emailErr &&
      !phoneErr
    )
      setIsSubmit(true);
  }, [customerFields]);

  return (
    <>
      <div className="sliding-panel-container">
        <div className="sliding-panel">
          <div className="title-row opp-header-text">
            <button type="button" className="mob-steps-back" onClick={closeAction}>
              <img src={ImageConfig.CHEVRON_LEFT} alt="Back" />
            </button>
            {i18n.t('addCustomer')}
            <button className="panel-close-icon link-anchor-button" onClick={closeAction} type="button">
              <img src={ImageConfig.CLOSE_BTN} alt="Close" />
            </button>
          </div>
          <div className="all-opportunity-steps-container pt-24">
            <div className="opportunity-forms">
              <p className="add-subtitle">{i18n.t('customerInfo')}</p>
              <div className="">
                <div className="steps-one-forms">
                  <form>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="name">
                        {i18n.t('customerName')}
                      </label>
                      <input
                        type="text"
                        className="form-control"
                        placeholder={i18n.t('giveCustomerName')}
                        id="name"
                        onChange={onInputValueChange}
                        onBlur={validateField}
                      />
                    </div>
                    <p className="add-subtitle">{i18n.t('contactAddr')}</p>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="addressLine1">
                        {i18n.t('addr')}
                      </label>
                      <input
                        type="text"
                        className="form-control"
                        placeholder={i18n.t('giveAddr')}
                        id="addressLine1"
                        onChange={onInputValueChange}
                        onBlur={validateField}
                      />
                    </div>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="country">
                        {i18n.t('country')}
                      </label>
                      <select className="form-control iptor-dd" id="country" onChange={onInputValueChange} onBlur={validateField}>
                        <option disabled selected>
                          {i18n.t('selectCountry')}
                        </option>
                        {state.enviornmentConfigs.crmCountryInfo.map((obj: CountryInfo) => {
                          return <option value={obj.country}>{obj.description}</option>;
                        })}
                      </select>
                    </div>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="area">
                        {i18n.t('area')}
                      </label>
                      <select className="form-control iptor-dd" id="area" onChange={onInputValueChange} onBlur={validateField}>
                        <option disabled selected>
                          {i18n.t('selectArea')}
                        </option>
                        {state.enviornmentConfigs.crmAreaInfo.map((obj: AreaInfo) => {
                          return (
                            <option value={obj.area}>
                              {obj.area} - {obj.description}
                            </option>
                          );
                        })}
                      </select>
                    </div>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="phone">
                        {i18n.t('phone')}
                      </label>
                      <input
                        type="text"
                        className="form-control"
                        placeholder={i18n.t('giveNumber')}
                        id="phone"
                        onChange={onInputValueChange}
                        onBlur={validateField}
                      />
                      {phoneErr && <span className="form-hints">{i18n.t('numericFieldError')}</span>}
                    </div>
                    <p className="add-subtitle">{i18n.t('accountInfo')}</p>
                    <div className="form-group oppty-form-elements">
                      <label className="cust-label" htmlFor="EMAIL">
                        {i18n.t('email')}
                      </label>
                      <input
                        type="text"
                        className="form-control"
                        placeholder={i18n.t('giveEmail')}
                        id="EMAIL"
                        onChange={onInputValueChange}
                        onBlur={validateField}
                      />
                      {emailErr && <span className="form-hints">{i18n.t('emailFieldError')}</span>}
                    </div>
                  </form>
                </div>
                <div className="step-nextbtn-with-arrow stepsone-nxtbtn">
                  <button className={isSubmit ? 'customer-btn' : 'customer-btn disable-link'} onClick={isSubmit ? onSubmit : undefined} type="button">
                    {!!state.auth.user.role && state.auth.user.role === 'Admin' ? i18n.t('addCustomerBtn') : i18n.t('addShareApproval')}
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default AddCustomer;
