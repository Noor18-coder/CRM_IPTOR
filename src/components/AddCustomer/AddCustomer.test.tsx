import axios from 'axios';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import AddCustomer from './AddCustomer';
import StoreMock from '../../mocks/Store.mock';
import { AppState } from '../../store';
import CustomerDetailsMock from '../../mocks/CustomerDetails.mock';
import ConfigMock from '../../mocks/Config.mock';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('[AddCustomer] AddCustomer', () => {
  let oGetElementById: any;
  beforeAll(() => {
    oGetElementById = document.getElementById;
  });
  afterAll(() => {
    document.getElementById = oGetElementById;
  });

  it('should renders correctly', () => {
    const store = StoreMock.createInitialStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddCustomer />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change value of input', async () => {
    document.getElementById = jest.fn().mockImplementation((id) => ({ id, style: { border: '' } }));

    const initSt = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initSt,
      enviornmentConfigs: {
        ...initSt.enviornmentConfigs,
        defaultOpprtunityInfo: { ...initSt.enviornmentConfigs.defaultOpprtunityInfo, currencyLDA: 'USD', language: 'en' },
        crmCountryInfo: ConfigMock.getCountryInfo(5),
        crmAreaInfo: ConfigMock.getAreaInfo(5),
      },
      auth: { ...initSt.auth, user: { ...initSt.auth.user, role: 'Admin' } },
    };

    const store = StoreMock.createAnyStore<AppState>(state);
    const responseData = CustomerDetailsMock.getAddBusinessPartnerResponse();
    mockedAxios.post.mockResolvedValue({ data: responseData, status: 200 });

    const wrapper = mount(
      <Provider store={store}>
        <AddCustomer />
      </Provider>
    );

    expect(wrapper.contains(<AddCustomer />)).toBe(true);

    const nameInp = wrapper.find('input#name');
    const addressLine1Inp = wrapper.find('input#addressLine1');
    const countryInp = wrapper.find('select#country');
    const EMAILInp = wrapper.find('input#EMAIL');
    const areaInp = wrapper.find('select#area');
    const phoneInp = wrapper.find('input#phone');
    const addCustomerBtn = wrapper.find('button#addCustomerBtn');

    (nameInp.instance() as any).value = 'Some Name';
    (addressLine1Inp.instance() as any).value = 'Street Address';
    (EMAILInp.instance() as any).value = 'some@gmail.com';
    (phoneInp.instance() as any).value = '1234567890';

    (countryInp.find('option').at(3).instance() as any).selected = true;
    (areaInp.find('option').at(3).instance() as any).selected = true;

    nameInp.simulate('change').simulate('blur');
    addressLine1Inp.simulate('change').simulate('blur');
    countryInp.simulate('change').simulate('blur');
    EMAILInp.simulate('change').simulate('blur');
    areaInp.simulate('change').simulate('blur');
    phoneInp.simulate('change').simulate('blur');

    expect(nameInp.hasClass('form-control')).toBe(true);
    expect(addressLine1Inp.hasClass('form-control')).toBe(true);
    expect(countryInp.hasClass('form-control')).toBe(true);
    expect(EMAILInp.hasClass('form-control')).toBe(true);
    expect(areaInp.hasClass('form-control')).toBe(true);
    expect(phoneInp.hasClass('form-control')).toBe(true);

    expect(addCustomerBtn.hasClass('disable-link')).toBe(true);
  });

  it('should change value of input on blur', async () => {
    document.getElementById = jest.fn().mockImplementation((id) => ({ id, style: { border: '' } }));

    const initSt = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initSt,
      enviornmentConfigs: {
        ...initSt.enviornmentConfigs,
        defaultOpprtunityInfo: { ...initSt.enviornmentConfigs.defaultOpprtunityInfo, currencyLDA: 'USD', language: 'en' },
        crmCountryInfo: ConfigMock.getCountryInfo(5),
        crmAreaInfo: ConfigMock.getAreaInfo(5),
      },
      auth: { ...initSt.auth, user: { ...initSt.auth.user, role: 'Admin' } },
    };

    const store = StoreMock.createAnyStore<AppState>(state);
    const responseData = CustomerDetailsMock.getAddBusinessPartnerResponse();
    mockedAxios.post.mockResolvedValue({ data: responseData, status: 200 });

    const wrapper = mount(
      <Provider store={store}>
        <AddCustomer />
      </Provider>
    );

    expect(wrapper.contains(<AddCustomer />)).toBe(true);

    const nameInp = wrapper.find('input#name');
    const addressLine1Inp = wrapper.find('input#addressLine1');
    const countryInp = wrapper.find('select#country');
    const EMAILInp = wrapper.find('input#EMAIL');
    const areaInp = wrapper.find('select#area');
    const phoneInp = wrapper.find('input#phone');
    const addCustomerBtn = wrapper.find('button#addCustomerBtn');

    (nameInp.instance() as any).value = '';
    (addressLine1Inp.instance() as any).value = '';
    (EMAILInp.instance() as any).value = '';
    (phoneInp.instance() as any).value = '';

    (countryInp.find('option').at(0).instance() as any).selected = true;
    (areaInp.find('option').at(0).instance() as any).selected = true;

    nameInp.simulate('change').simulate('blur');
    addressLine1Inp.simulate('change').simulate('blur');
    countryInp.simulate('change').simulate('blur');
    EMAILInp.simulate('change').simulate('blur');
    areaInp.simulate('change').simulate('blur');
    phoneInp.simulate('change').simulate('blur');

    expect(nameInp.hasClass('form-control')).toBe(true);
    expect(addressLine1Inp.hasClass('form-control')).toBe(true);
    expect(countryInp.hasClass('form-control')).toBe(true);
    expect(EMAILInp.hasClass('form-control')).toBe(true);
    expect(areaInp.hasClass('form-control')).toBe(true);
    expect(phoneInp.hasClass('form-control')).toBe(true);

    expect(addCustomerBtn.hasClass('disable-link')).toBe(true);
  });

  it('should not validate email on change', async () => {
    const style = { border: '' };
    document.getElementById = jest.fn().mockImplementation((id) => ({ id, style }));
    const docSpy = jest.spyOn(document, 'getElementById');

    const initSt = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initSt,
      enviornmentConfigs: {
        ...initSt.enviornmentConfigs,
        defaultOpprtunityInfo: { ...initSt.enviornmentConfigs.defaultOpprtunityInfo, currencyLDA: 'USD', language: 'en' },
        crmCountryInfo: ConfigMock.getCountryInfo(5),
        crmAreaInfo: ConfigMock.getAreaInfo(5),
      },
      auth: { ...initSt.auth, user: { ...initSt.auth.user, role: 'Admin' } },
    };

    const store = StoreMock.createAnyStore<AppState>(state);

    const wrapper = mount(
      <Provider store={store}>
        <AddCustomer />
      </Provider>
    );

    expect(wrapper.contains(<AddCustomer />)).toBe(true);

    const EMAILInp = wrapper.find('input#EMAIL');
    const EMAILInpInstance = EMAILInp.instance() as any;
    EMAILInpInstance.value = 'someemail';
    EMAILInp.simulate('change');

    expect(EMAILInp.hasClass('form-control')).toBe(true);
    expect(style.border).toBe('1px solid #ED2024');
    expect(docSpy).toHaveBeenCalledWith('EMAIL');
  });

  it('should not validate phone on change', async () => {
    const style = { border: '' };
    document.getElementById = jest.fn().mockImplementation((id) => ({ id, style }));
    const docSpy = jest.spyOn(document, 'getElementById');

    const initSt = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initSt,
      enviornmentConfigs: {
        ...initSt.enviornmentConfigs,
        defaultOpprtunityInfo: { ...initSt.enviornmentConfigs.defaultOpprtunityInfo, currencyLDA: 'USD', language: 'en' },
        crmCountryInfo: ConfigMock.getCountryInfo(5),
        crmAreaInfo: ConfigMock.getAreaInfo(5),
      },
      auth: { ...initSt.auth, user: { ...initSt.auth.user, role: 'Admin' } },
    };

    const store = StoreMock.createAnyStore<AppState>(state);

    const wrapper = mount(
      <Provider store={store}>
        <AddCustomer />
      </Provider>
    );

    expect(wrapper.contains(<AddCustomer />)).toBe(true);

    const phoneInp = wrapper.find('input#phone');
    const phoneInpInstance = phoneInp.instance() as any;
    phoneInpInstance.value = 'ssjasjh';
    phoneInp.simulate('change');

    expect(phoneInp.hasClass('form-control')).toBe(true);
    expect(style.border).toBe('1px solid #ED2024');
    expect(docSpy).toHaveBeenCalledWith('phone');
  });
});
