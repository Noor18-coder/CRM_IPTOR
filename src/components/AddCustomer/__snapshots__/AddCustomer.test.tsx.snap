// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AddCustomer] AddCustomer should renders correctly 1`] = `
<div
  className="sliding-panel-container fixed-header-footer"
>
  <div
    className="sliding-panel"
  >
    <div
      className="title-row opp-header-text"
    >
      <button
        className="mob-steps-back"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Back"
          src="chevron-left-black.svg"
        />
      </button>
      Add New Customer
      <button
        className="panel-close-icon link-anchor-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="all-opportunity-steps-container"
    >
      <div
        className="opportunity-edit-forms"
      >
        <div
          className="opportunity-forms edit-form"
        >
          <p
            className="add-subtitle"
          >
            Customer Information
          </p>
          <div
            className=""
          >
            <div
              className="steps-one-forms"
            >
              <form>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="name"
                  >
                    Customer Name
                  </label>
                  <input
                    className="form-control"
                    id="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Give customer a name"
                    type="text"
                  />
                </div>
                <p
                  className="add-subtitle"
                >
                  Contact Address
                </p>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="addressLine1"
                  >
                    Address
                  </label>
                  <input
                    className="form-control"
                    id="addressLine1"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Give address"
                    type="text"
                  />
                </div>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="country"
                  >
                    Country
                  </label>
                  <select
                    className="form-control iptor-dd"
                    id="country"
                    onBlur={[Function]}
                    onChange={[Function]}
                  >
                    <option
                      disabled={true}
                      selected={true}
                    >
                      Select country
                    </option>
                  </select>
                </div>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="area"
                  >
                    Area
                  </label>
                  <select
                    className="form-control iptor-dd"
                    id="area"
                    onBlur={[Function]}
                    onChange={[Function]}
                  >
                    <option
                      disabled={true}
                      selected={true}
                    >
                      Select area
                    </option>
                  </select>
                </div>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="phone"
                  >
                    Phone Number
                  </label>
                  <input
                    className="form-control"
                    id="phone"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Give phone number"
                    type="text"
                  />
                </div>
                <p
                  className="add-subtitle"
                >
                  Account Information
                </p>
                <div
                  className="form-group oppty-form-elements"
                >
                  <label
                    className="cust-label"
                    htmlFor="EMAIL"
                  >
                    Email Address
                  </label>
                  <input
                    className="form-control"
                    id="EMAIL"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Give email address"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          className="step-nextbtn-with-arrow stepsone-nxtbtn"
        >
          <button
            className="customer-btn disable-link"
            id="addCustomerBtn"
            type="button"
          >
            Add & Share For Approval
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
